### ISSUE TITLE
Bidirectional dataflow?

### ISSUE THREAD

**Issue by harsh02** [OWASP/pytm#258]
I am trying to create bidirectional dataflow with single definition and on line pointing arrow both sides, any way I can achieve this? <--->

**Comment by izar** [comment]
Not really, since that would break the (source, destination, data) definition of a dataflow that we have in order to be able to identify threats on dataflows.

**Comment by raphaelahrens** [comment]
What I used sometimes is the this custom function

```python
def req_reply(src, dest, name, reply_args=None, **kwargs):
    if not reply_args:
        reply_args = {
            'name': gen_reply_name(name)
        }
    req = Dataflow(src, dest, name, **kwargs)
    reply = Dataflow(dest, src, **reply_args)
    reply.responseTo = req
    return req, reply
```

and use it like this

```python
req_reply(a, b, 'Flow description')
req_reply(a, b, 'Flow description',  repy_args={name: "custom reply description"})

```

If you then add `mergeResponses` into your `TM` it even merges the arrows into one.
```python
tm = TM(
            `my tm`,
            mergeResponses=True,
)
```

**Comment by izar** [comment]
Interesting. You should PR that in...

**Comment by raphaelahrens** [comment]
I wonder if it would be useful to have the ability to "compress" dataflows into a single arrow for a high level view.
For example some things like a login have a typical flow of messages and could be reused in multiple models, but for a higher view it could be enough to just see a fat arrow for login.

The mechanics would be very similar to `mergeResponses`.

Maybe this could then also be used in @harsh02 case.

**Comment by harsh02** [comment]
Thanks guys!

**Comment by harsh02** [comment]
@raphaelahrens I am using your custom function and working great, I am just trying to fix one side arrow being ended up out scope and greyed out. Did you run into same issue? I have defined assets in scope.

**Comment by raphaelahrens** [comment]
No can you share a short test model where this happens?

**Comment by harsh02** [comment]
DFD look like this

![image](https://github.com/user-attachments/assets/0c53c44a-2fd4-436d-afee-7469c7f29585)

Definition looks like this

![image](https://github.com/user-attachments/assets/7816b0b5-73e8-4c7e-a827-1d696ece307e)

**Comment by raphaelahrens** [comment]
I'm not sure what the problem is. Do you mean that one arrow and has a white fill color?
It would be helpful to see what you would expect and what you get.
Currently this looks like the normal behavior.

**Comment by harsh02** [comment]
Yes, to my thinking one arrow being white filled means out of scope? But I could be wrong too. I was expecting both side arrows to be black filled.

**Comment by raphaelahrens** [comment]
Ah OK, no that just means that the white arrow is a reply to the black arrow.

**Comment by raphaelahrens** [comment]
Can this be closed?
